onboarding/source/amb_light_service.c

```diff
@@ -18,22 +28,89 @@ static void lightServiceTask(void * pvParameters);
 uint8_t initLightService(void) {
     /* USER CODE BEGIN */
     // Create the task and queue here.
+    BaseType_t xReturned = pdFAIL;
+    if(lightServiceTaskHandle == NULL){
+        xReturned = xTaskCreate(lightServiceTask,
+                                LIGHT_SERVICE_NAME,
+                                LIGHT_SERVICE_STACK_SIZE,
+                                NULL,
+                                LIGHT_SERVICE_PRIORITY,
+                                &lightServiceTaskHandle);
 
+    }
+
+    if(xReturned == pdFAIL){
+        printf("Unable to create a light service task\n");
+    }
+	
+    if(xLightServiceQueue == NULL){
+        xLightServiceQueue = xQueueCreate(QUEUE_SiZE, LIGHT_EVENT_SIZE);
+    }
     /* USER CODE END */
     return 1;
 }
 
+/**
+ * @brief Get light data from adcData
+ * @param void
+ **/
+uint16_t Get_Light_Sensor_data(void)
+{
+
+	adcData_t *adc_data_ptr = pvPortMalloc(sizeof(adcData_t));
+    adc_data_ptr->id = 6;
+    //adc_data_ptr->value;
+
+ 	/** - Start Group1 ADC Conversion 
+ 	*     Select Channel 6 - Light Sensor for Conversion
+ 	*/
+	adcStartConversion(adcREG1, adcGROUP1);
+
+ 	/** - Wait for ADC Group1 conversion to complete */
+ 	while(!adcIsConversionComplete(adcREG1, adcGROUP1)); 
+
+	/** - Read the conversion result
+	*     The data contains the Light sensor data
+    */
+	adcGetData(adcREG1, adcGROUP1, adc_data_ptr);
+	
+	/** - Transmit the Conversion data to PC using SCI
+    */
+	return (adc_data_ptr->value);
+}
+
 static void lightServiceTask(void * pvParameters) {
     /* USER CODE BEGIN */
     // Wait for MEASURE_LIGHT event in the queue and then print the ambient light value to the serial port.
-
+    ASSERT(lightServiceTaskHandle != NULL);
+    light_event_t queueTask;
+	
+    while (xQueueReceive(xLightServiceQueue, &queueTask, 0) == pdTRUE) {
+	    
+        if (queueTask == MEASURE_LIGHT){
+		
+        	uint16 Light_Sensor_Data = Get_Light_Sensor_data();
+        	int length = snprintf(NULL, 0, "%c;", (char)Light_Sensor_Data);
+        	char *data = malloc(length + 1);
+        
+        	sprintf(data, "%c;", (char)Light_Sensor_Data);
+        	sciPrintText(scilinREG, (unsigned char*)data, length + 1);
+        
+        	free(data);
+	}
+    }
     /* USER CODE END */
 }
 
-uint8_t sendToLightServiceQueue(light_event_t *event) {
+/**
+ * @brief Send a light service to the queue
+ * @param light_event_t *event
+ **/
+uint8_t sendToLightServiceQueue(light_event_t *event) { 
     /* USER CODE BEGIN */
     // Send the event to the queue.
-    
```

Comment:
Check if `event` is NULL. In general, if you have a function parameter that's a pointer, then you should be checking if it's NULL inside the function.